dash_labs/dashdown.py

```diff
@@ -0,0 +1,419 @@
+     a separate file so you can run it individually. We could integrate jinja in here perhaps…: {% include code.py %}
+    - Add markdown files to hot reload in dash.  That way users can have the same hot-reloading dev experience when working in markdown
+    - if code block is not executed don't register callbacks and layout
+    - Need to remove if __name__ == "__main__": ...   from the code blocks?
+    - refactor the _update_props function.
+
+"""
+
+from dash import dcc, html, dash_table, Input, Output, State, callback
+import re
+import ast
+import copy
+from jinja2 import Template
+import plotly.express as px
+import plotly
+import dash_bootstrap_components as dbc
+import warnings
+
+
+def warning_message(message, category, filename, lineno, line=None):
+    return f"{category.__name__}:\n {message} \n"
+
+
+warnings.formatwarning = warning_message
+
+
+class MarkdownAIO(html.Div):
+    def __init__(
+        self,
+        filename,
+        scope=None,
+        scope_creep=False,
+        dash_scope=True,
+        template_variables=None,
+        exec_code=False,
+        side_by_side=False,
+        code_first=True,
+        code_markdown_props={},
+        text_markdown_props={},
+        clipboard_props={},
+        app_div_props={},
+    ):
+        """
+        MarkdownAIO is an All-In-One component to display content of Markdown files with the option to run and/or
+         display code blocks.
+
+        - `filename`(string):
+           The path to the markdown file.
+
+        - `scope`(dict):
+           Add scope to the code blocks. When using `app.callback` the `app` must be included.
+           scope=dict(app=app)
+
+        - `scope_creep`(boolean; default False):
+           Allows variables from one code block to be defined in the next code block.
+
+        - `dash_scope`(boolean; default True):
+           If True, the default scope is:
+            ```scope = dict(
+                dcc=dcc,
+                html=html,
+                Input=Input,
+                Output=Output,
+                State=State,
+                dash_table=dash_table,
+                px=px,
+                plotly=plotly,
+                dbc=dbc,
+                **(scope or {}),
+            )```
+
+        - `template_variables` (dict):
+           Variable passed to the  jinja templating engine: https://jinja.palletsprojects.com/en/3.0.x/templates/
+           This is a way to display dynamic content.  For example:
+           `template_variables={‘language’: ‘english’})`
+           See the jinja docs for how to use the template variables in the markdown files.
+           `{% if language == 'english' %} Hello {% elif language == 'french' %} Bonjour {% endif %}`
+
+        - `exec_code` (boolean; default False):
+           If `True`, code blocks will be executed.  This may also be set within the code block with the comment
+            # exec-code-true or # exec-code-false
```

Comment:
It can actually be anywhere in the code block on a comment line.  It's just that the text won't be won't be displayed if they are included on the first line with the back ticks.  Since there are 7 props that this applies to, maybe it would be better to have the full example in the docs instead of here.